### PR TITLE
Client sweep automation

### PR DESCRIPTION
**Pull request by dhvanichheda**
Changes for client sweep automation feature.

### CODE REVIEW

**Review by sushmakt — COMMENTED** [review]
Both Table.php and ClientSweepAutomation.php being new files seem to be inconsistent with the code convention followed by the rest of files.
1. Namely, a space after class and function definitions before the "{" and uniform spacing between two functions. Some function seems to have no space vs others.
2. Also, please add the Intel license header for these new files.

File: base/ClientSweepAutomation.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        $this->failedReq = 0;
+        $this->diff_percent = 0;
+        $this->plateau_tracker = 0;
+        $this->clientsweep_results = array();
+        $this->optimalClientThreads = 0;
+    }
+
+    public function GetOptimalThreads(PerfOptions $options, PerfTarget $target){
+        $options->clientThreads = 0;
+        for($iter=1; $iter<=self::client_sweep_max_iterations; $iter++){
```
Comment (sushmakt) [inline base/ClientSweepAutomation.php]: Please change iter to something more meaningful since it's used throughout the file.

**Review by sushmakt — CHANGES_REQUESTED** [review]
Variable naming in ClientSweepAutomation

File: base/ClientSweepAutomation.php (hunk 1 above)
Comment (dhvanichheda) [reply base/ClientSweepAutomation.php]: Done

File: base/ClientSweepAutomation.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      return $this->optimalClientThreads;
+    }
+
+    private function ParseSiegeOutput(Siege $siege, PerfOptions $options, $iteration) {
+      $siege_stats = $siege->collectStats();
+      $this->curr_tps = $siege_stats['Combined']['Siege RPS'];
+      $this->failedReq = $siege_stats['Combined']['Siege failed requests'];
+      if ($iteration != 1) {
+        $this->diff_percent = (((floor($this->curr_tps - $this->prev_tps))/$this->prev_tps)*100);
+        $this->diff_percent = intval($this->diff_percent * ($p = pow(10,2))) / $p;
```
Comment (dychen95) [inline base/ClientSweepAutomation.php]: Is this just for formatting the number? If so, how about use number_format()?

File: base/ClientSweepAutomation.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        "TPS" => $this->curr_tps,
+        "diffPercentage" => $this->diff_percent . " %",
+        "failedRequests" => $this->failedReq
+      );
+      array_push($this->clientsweep_results, $row);
+    }
+
+    private function CheckIfContinue(PerfOptions $options, $iteration) {
+      if ($this->failedReq > 0) {
+          return false;
```
Comment (dychen95) [inline base/ClientSweepAutomation.php]: indentation is not consistent. Some have 2-space indentation and some have 4.

File: base/ClientSweepAutomation.php (hunk 2 above)
Comment (dhvanichheda) [reply base/ClientSweepAutomation.php]: Yes it is just for formatting. I will change it and use number_format(). I hadn't come across this function before but it does seem more readable.

File: base/ClientSweepAutomation.php (hunk 3 above)
Comment (dhvanichheda) [reply base/ClientSweepAutomation.php]: Okay I will take care of that.

**Review by dychen95 — APPROVED** [review]
LGTM

**Comment by sushmakt** [comment]
Merged.

**Pull request merged**